### PR TITLE
 Replace Tooltip test case with testing library

### PR DESCRIPTION
**Pull request by zpc7**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |       -    |
| 🇨🇳 Chinese |       -    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by zombieJ** [comment]
Coverage 降了

File: components/tooltip/__tests__/tooltip.test.js [hunk 1]
```diff
@@ -256,122 +272,101 @@ describe('Tooltip', () => {
...
-      mount(
+      render(
         <Tooltip title={0} visible autoAdjustOverflow={undefined}>
           <div />
         </Tooltip>,
       );
     }).not.toThrow();
   });
 
-  it('support other placement', done => {
```
Comment (zombieJ) [inline components/tooltip/__tests__/tooltip.test.js:299]: 这个不能删

File: components/tooltip/__tests__/tooltip.test.js [hunk 2]
```diff
@@ -256,122 +272,121 @@ describe('Tooltip', () => {
...
-        mouseEnterDelay={0}
-        ref={ref}
-      >
-        <span>Hello world!</span>
-      </Tooltip>,
-    );
+        expect(container.getElementsByTagName('span')).toHaveLength(1);
+        const element = container.getElementsByTagName('span')[0];
+        fireEvent.mouseEnter(element);
+        await sleep(500);
```
Comment (zombieJ) [inline components/tooltip/__tests__/tooltip.test.js:349]: 这个可以尝试改成  jest.runAllTimer 吗？切换成 testing lib 后理论上都可以被 act 了

Comment (zpc7) [reply components/tooltip/__tests__/tooltip.test.js:349]: 尝试: https://github.com/ant-design/ant-design/pull/35396/commits/ad886d1f65d83a92bc9810d6d74c71a3dc1cc608
发现在 React16/17 下, 测试会挂掉

**Comment by hengkx** [comment]
这个没有 squash？

**Pull request merged**